src/ComparisonResult.ts

```diff
@@ -0,0 +1,37 @@
+/*!
+ * Copyright 2012 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+
+/** Comparison result types for the comparison report */
+export enum ComparisonResultType {
+    EQUAL = "EQUAL",
+    NOT_EQUAL = "NOT_EQUAL",
+    ERROR = "ERROR"
```

Comment:
Could you add some comments explaining what each of these enum variants mean? (In particular, what does `ERROR` mean and when is it different from `NOT_EQUAL`?)